### PR TITLE
[feature] Document Authorization and Rate Limiting [OSF-8172][OSF-8068]

### PR DESCRIPTION
**Pull request by caseyrollins**
#### Purpose
- Document authentication.
- Add a note about rate limits.
- Some alphabetization and file renaming.

#### Tickets
- [OSF-8172](https://openscience.atlassian.net/browse/OSF-8172)
- [OSF-8068](https://openscience.atlassian.net/browse/OSF-8068)

### CODE REVIEW

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
For OSF-8068, we really need a description of dealing with rate limiting when they get it as well. Just a note about how to find out how long they're throttled for and that they have to wait for that limit to pass.

**Comment by caseyrollins** [comment]
Ah, I figured there was more to it. I'll add that!

File: swagger-spec/introduction/error_codes.yaml [hunk 1]
```diff
@@ -76,7 +76,13 @@ get:
     The user has sent too many requests in a given amount of time.
 
 
-    Some actions are more limited than others, but in general, send [authenticated](#Introduction_authentication) requests for a much higher rate limit.
+    Wait at least the amount of time specified in the `Retry-After` header of the response before sending another request.
```
Comment (caseyrollins) [inline swagger-spec/introduction/error_codes.yaml]: @brianjgeiger, better for OSF-8068?

File: swagger-spec/introduction/authentication.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    #### OAuth
+
+    The OSF allows third-party web applications to connect to the OSF on behalf of other users via the OAuth 2.0 web application flow.
+
+    You can add a developer application from the OSF settings page (either on the [OSF](https://osf.io/settings/tokens/) or our [testing server](https://test.osf.io/settings/tokens/)).
+
+
+    We will be adding narrative documentation for handling OAuth flows, but for now, if you're familiar with OAuth, you can check out the [CAS overlay documentation](https://github.com/CenterForOpenScience/cas-overlay#web-server-authorization) for options.
+
+    We've also created a [test application](https://github.com/abought/osf_oauth2_demo) for verifying that our OAuth workflow works correctly.
```
Comment (brianjgeiger) [inline swagger-spec/introduction/authentication.yaml:23]: We should make a ticket to bring this in-house.

**Review by brianjgeiger — APPROVED** [review]

File: swagger-spec/introduction/authentication.yaml (hunk 2 above)
Comment (caseyrollins) [reply swagger-spec/introduction/authentication.yaml:23]: Agreed. Fine to leave in for now though?

Comment (brianjgeiger) [reply swagger-spec/introduction/authentication.yaml:23]: Yup. This PR is approved

**Pull request merged**